qadence/qinfo_tools/qfi.py

```diff
@@ -0,0 +1,489 @@
+import numpy as np
+import torch
+from torch import Tensor
+from torch.autograd import grad
+
+from qadence import Overlap, OverlapMethod
+from qadence.circuit import QuantumCircuit
+from qadence.backend import BackendName
+from qadence.measurements import Measurements
+from qadence import DiffMode
+from qadence.blocks import primitive_blocks, parameters
+
+
+def symsqrt(a, cond=None, return_rank=False):
+    """Computes the symmetric square root of a positive definite matrix.
+    Taken from https://github.com/pytorch/pytorch/issues/25481"""
+
+    s, u = torch.linalg.eigh(a)
+    cond_dict = {
+        torch.float32: 1e3 * 1.1920929e-07,
+        torch.float64: 1e6 * 2.220446049250313e-16,
+    }
+
+    if cond in [None, -1]:
+        cond = cond_dict[a.dtype]
+
+    above_cutoff = abs(s) > cond * torch.max(abs(s))
+
+    psigma_diag = torch.sqrt(s[above_cutoff])
+    u = u[:, above_cutoff]
+
+    B = u @ torch.diag(psigma_diag) @ u.t()
+    if return_rank:
+        return B, len(psigma_diag)
+    else:
+        return B
+
+
+def _overlap_with_inputs(
+    circuit: QuantumCircuit,
+    vparams: tuple | None = None,
+    backend: BackendName = BackendName.PYQTORCH,
```

Comment:
Have you tested what is the performance with the other backends and GPSR? I am asking out of curiousity.

Reply:
Not yet! I have only tested pyqtorch for the moment